src/Nova/Resource.php

```diff
@@ -4,11 +4,94 @@
 
 namespace Tipoff\Support\Nova;
 
+use Illuminate\Http\Request;
+use Laravel\Nova\Fields\BelongsTo;
+use Laravel\Nova\Fields\DateTime;
+use Laravel\Nova\Fields\Field;
+use Laravel\Nova\Fields\ID;
 use Laravel\Nova\Http\Requests\NovaRequest;
 use Laravel\Nova\Resource as NovaResource;
 
 abstract class Resource extends NovaResource
 {
+    protected array $filterClassList = [];
+
+    protected array $cardClassList = [];
+
+    protected array $lensClassList = [];
+
+    protected array $actionClassList = [];
+
+    /**
+     * @return array|Field[]
+     */
+    protected function dataFields(): array
+    {
+        return [
+            ID::make(),
+        ];
+    }
+
+    /**
+     * @return array|Field[]
+     */
+    protected function creatorDataFields(): array
+    {
+        $userResource = nova('user');
+
+        return $userResource ? [
+            BelongsTo::make('Created By', 'creator', $userResource)->exceptOnForms(),
+            DateTime::make('Created At')->exceptOnForms(),
+        ] : [];
+    }
+
+    /**
+     * @return array|Field[]
+     */
+    protected function updaterDataFields(): array
+    {
+        $userResource = nova('user');
+
+        return $userResource ? [
+            BelongsTo::make('Updated By', 'updater', $userResource)->exceptOnForms(),
+            DateTime::make('Updated At')->exceptOnForms(),
+        ] : [];
+    }
```

Comment:
In every Nova resource, I like to separate out the following into a panel at the bottom:

- ID field
- Timestamps
- Creator (if applicable)
- Updater (if applicable)

Example:

```
protected function dataFields()
    {
        return [
            ID::make(),
            BelongsTo::make('Created By', 'creator', \App\Nova\User::class)->exceptOnForms(),
            DateTime::make('Created At')->exceptOnForms(),
            BelongsTo::make('Updated By', 'updater', \App\Nova\User::class)->exceptOnForms(),
            DateTime::make('Updated At')->exceptOnForms(),
        ];
    }
```

Could we find a way to add that to this resource stub?

Reply:
You'll see the base class helpers are used in the derived classes to merge in the creator and updater fields when appropriate.

```
    protected function dataFields(): array
    {
        return array_merge(
            parent::dataFields(),
            $this->creatorDataFields(),
            $this->updaterDataFields(),
        );
    }
```

Reply:
Ok, thank you. Does this preserve the order with the timestamps?

Reply:
Yes -- in this example, the end result is identical.

Reply:
Thank you!